agent/tsunami_agent.py

```diff
@@ -25,20 +26,31 @@
 logger = logging.getLogger(__name__)
 
 
-def _prepare_domain_name(message: msg.Message) -> Union[str, None]:
+def _should_process_url(scope_urls_regex: Optional[str], url: str) -> bool:
+    if scope_urls_regex is None:
+        return True
+    link_in_scan_domain = re.match(scope_urls_regex, url) is not None
+    if not link_in_scan_domain:
+        logger.warning('link url %s is not in domain %s', url, scope_urls_regex)
+    return link_in_scan_domain
+
+
+def _prepare_domain_name_and_url(message: msg.Message, scope_urls_regex: Optional[str]) -> Union[str, None]:
```

Comment:
This logic is not correct and the function is here doing more that the name suggests.

The should process should force the processing to return, here it is returning None.

Prepare domain name and URL should not be checking should process here.